src/core_write.cpp

```diff
@@ -141,40 +142,41 @@ void ScriptToUniv(const CScript& script, UniValue& out, bool include_address)
     out.pushKV("asm", ScriptToAsmStr(script));
     out.pushKV("hex", HexStr(script));
 
-    std::vector<std::vector<unsigned char>> solns;
-    TxoutType type = Solver(script, solns);
-    out.pushKV("type", GetTxnOutputType(type));
-
+    TxoutType type;
     CTxDestination address;
-    if (include_address && ExtractDestination(script, address) && type != TxoutType::PUBKEY) {
+
+    if (ExtractDestination(script, type, address) && include_address && type != TxoutType::PUBKEY) {
         out.pushKV("address", EncodeDestination(address));
     }
+    out.pushKV("type", GetTxnOutputType(type));
```

Comment:
Why are you changing the order?

Reply:
This doesn't seem addressed: https://github.com/bitcoin/bitcoin/pull/20286/files#r547134259 (it's ok if you disagree with the comment, or there is a good reason to - but do respond if you can).

Reply:
The first thing done in `ExtractDestination` is
```c++
std::vector<valtype> vSolutions;
whichType = Solver(scriptPubKey, vSolutions);
```
so prior to this change, `Solver` code was duplicated and running twice.

By changing the order in the if statement from `include_address && ExtractDestination(script, address` => `ExtractDestination(script, type, address) && include_address` we are guaranteed to always have `type` set by `ExtractDestination`, and can get rid of the duplicated `Solver` code here.